azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -102,6 +102,17 @@ func resourceArmKubernetesCluster() *schema.Resource {
 							ValidateFunc: validate.KubernetesAgentPoolName,
 						},
 
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(containerservice.AvailabilitySet),
+								string(containerservice.VirtualMachineScaleSets),
+							}, true),
```

Comment:
We are trying to not be case insensitive for constants unless requires. as such:
```suggestion
							}, false),
```